src/libraries/System.Security.Cryptography/tests/X509Certificates/X500DistinguishedNameBuilderTests.cs

```diff
@@ -0,0 +1,452 @@
+        {
+            AssertBuilder(expectedHex, builder => builder.Add(oid, value, tag));
+        }
+
+        [Theory]
+        [MemberData(nameof(AddStringTheories))]
+        public static void Add_Oid_Success(string oid, string value, UniversalTagNumber? tag, string expectedHex)
+        {
+            AssertBuilder(expectedHex, builder => builder.Add(new Oid(oid, null), value, tag));
+        }
+
+        [Fact]
+        public static void Add_WithException_ValidState()
+        {
+            X500DistinguishedNameBuilder builder = new();
+
+            // Cause an exception to be raised, and handled.
+            AssertExtensions.Throws<ArgumentException>(
+                "emailAddress",
+                () => builder.AddEmailAddress("\u0411\u0430\u043d\u0430\u043d"));
+
+            // Make sure the handled exception didn't put the builder in an invalid state.
+            builder.AddEmailAddress("k@example.com");
+            AssertBuilder("301E311C301A06092A864886F70D010901160D6B406578616D706C652E636F6D", builder);
+        }
+
+        [Fact]
+        public static void Build_Empty()
+        {
+            AssertBuilder("3000", new X500DistinguishedNameBuilder());
+        }
+
+        [Fact]
+        public static void Build_UsableAfterBuild()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            builder.AddOrganizationName("GitHub");
+            AssertBuilder("3011310F300D060355040A0C06476974487562", builder);
+            builder.AddOrganizationalUnitName("PSE");
+            AssertBuilder("301F310C300A060355040B0C03505345310F300D060355040A0C06476974487562", builder);
+        }
+
+        [Fact]
+        public static void Build_KitchenSink()
+        {
+            const string ExpectedHex =
+                "3081B9" +
+                    "31133011060A0992268993F22C6401191603636F6D" +
+                    "31163014060A0992268993F22C6401191606676974687562" +
+                    "310C300A060355040B0C03505345" +
+                    "310F300D060355040A0C06476974487562" +
+                    "310B3009060355040613025553" +
+                    "3111300F06035504080C0856697267696E6961" +
+                    "3113301106035504070C0A416C6578616E64726961" +
+                    "3120301E06092A864886F70D01090116116B6576696E406578616D706C652E636F6D" +
+                    "3114301206035504030C0B4B6576696E204A6F6E6573";
+
+            AssertBuilder(ExpectedHex, builder =>
+            {
+                builder.AddCommonName("Kevin Jones");
+                builder.AddEmailAddress("kevin@example.com");
+                builder.AddLocalityName("Alexandria");
+                builder.AddStateOrProvinceName("Virginia");
+                builder.AddCountryOrRegion("US");
+                builder.AddOrganizationName("GitHub");
+                builder.AddOrganizationalUnitName("PSE");
+                builder.AddDomainComponent("github");
+                builder.AddDomainComponent("com");
```

Comment:
I was about to say "it looks like we're missing the test that shows the same attribute type ID can be used more than once", but KitchenSink adds DC twice, so it's covered.